cluster/reconcile.go

```diff
@@ -0,0 +1,134 @@
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/rancher/rke/hosts"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/services"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/client-go/kubernetes"
+)
+
+func ReconcileCluster(kubeCluster, currentCluster *Cluster) error {
+	logrus.Infof("[reconcile] Reconciling cluster state")
+	if currentCluster == nil {
+		logrus.Infof("[reconcile] This is newly generated cluster")
+
+		return nil
+	}
+	// to handle if current local admin is down and we need to use new cp from the list
+	if !isLocalConfigWorking(kubeCluster.LocalKubeConfigPath) {
+		if err := rebuildLocalAdminConfig(kubeCluster); err != nil {
+			return err
+		}
+	}
+
+	kubeClient, err := k8s.NewClient(kubeCluster.LocalKubeConfigPath)
+	if err != nil {
+		return fmt.Errorf("Failed to initialize new kubernetes client: %v", err)
+	}
+
+	key, _ := checkEncryptedKey(kubeCluster.SSHKeyPath)
+
+	if err := reconcileWorker(currentCluster, kubeCluster, key, kubeClient); err != nil {
+		return err
+	}
+
+	if err := reconcileControl(currentCluster, kubeCluster, key, kubeClient); err != nil {
+		return err
+	}
+	logrus.Infof("[reconcile] Reconciled cluster state successfully")
+	return nil
+}
+
+func reconcileWorker(currentCluster, kubeCluster *Cluster, key ssh.Signer, kubeClient *kubernetes.Clientset) error {
+	// worker deleted first to avoid issues when worker+controller on same host
+	logrus.Infof("[reconcile] Check worker hosts to be deleted")
```

Comment:
I don't think we should log the `check` statements in INFO, may be the debug would be more appropriate